internal/controller/variantautoscaling_controller.go

```diff
@@ -592,3 +648,45 @@ func (r *VariantAutoscalingReconciler) readOptimizationConfig(ctx context.Contex
 	interval = cm.Data["GLOBAL_OPT_INTERVAL"]
 	return interval, nil
 }
+
+// handleServiceMonitorEvent handles events for the controller's own ServiceMonitor.
+// When ServiceMonitor is deleted, it logs a critical warning and emits a Kubernetes event.
+// This ensures that administrators are aware when the ServiceMonitor that enables
+// Prometheus scraping of controller metrics (including optimized replicas) is missing.
+func (r *VariantAutoscalingReconciler) handleServiceMonitorEvent(ctx context.Context, obj client.Object) []reconcile.Request {
+	serviceMonitor, ok := obj.(*unstructured.Unstructured)
+	if !ok {
+		return nil
+	}
+
+	name := serviceMonitor.GetName()
+	namespace := serviceMonitor.GetNamespace()
+
+	// Check if ServiceMonitor is being deleted
+	if !serviceMonitor.GetDeletionTimestamp().IsZero() {
+		logger.Log.Errorw("CRITICAL: ServiceMonitor being deleted - Prometheus will not scrape controller metrics",
```

Comment:
we are already logging with level error, I dont think we need word `CRITICAL` in logger